pkg/collection/collection.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package collection provides high-level abstractions for collections of files
+package collection
+
+import (
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type MetadataType uint8
+
+const (
+	FilenameMimeType MetadataType = iota
```

Comment:
Should this be `FilenameMetadataType`, or even `MetadataTypeFilename` just as for example http.Status* constants?

Also, why would this be needed now, as it is not used? I think that we should not add code in pr that is not directly usable within the scope of pr. We can add it later if needed, but for now there is only one metadata without any association to its MetadataType.